substrate/frame/transaction-storage/src/lib.rs

```diff
@@ -432,8 +446,8 @@ pub mod pallet {
 			let byte_fee = ByteFee::<T>::get().ok_or(Error::<T>::NotConfigured)?;
 			let entry_fee = EntryFee::<T>::get().ok_or(Error::<T>::NotConfigured)?;
 			let fee = byte_fee.saturating_mul(size.into()).saturating_add(entry_fee);
-			ensure!(T::Currency::can_slash(&sender, fee), Error::<T>::InsufficientFunds);
-			let (credit, _) = T::Currency::slash(&sender, fee);
+			T::Currency::hold(&HoldReason::StorageFeeHold.into(), &sender, fee)?;
+			let (credit, _) = T::Currency::slash(&HoldReason::StorageFeeHold.into(), &sender, fee);
```

Comment:
nit: `T::Currency::hold` must ensure there is no slash remainder, but just for tests
```suggestion
			let (credit, remainder) = T::Currency::slash(&HoldReason::StorageFeeHold.into(), &sender, fee);
			debug_assert!(remainder.is_zero());
```